### PR TITLE
Add more RenderEditable test coverage

### PR DESCRIPTION
**Pull request by xster**
https://github.com/flutter/flutter/issues/26026

- Moved the widget test out of the rendering layer because it prevents mixing rendering tests https://github.com/flutter/flutter/issues/27002
- editable_text_test.dart was becoming unwieldy. Moved cursor related tests to another file.
- Mainly for catching things missed in https://coveralls.io/builds/21244453/source?filename=lib/src/rendering/editable.dart

### CODE REVIEW

**Comment by xster** [comment]
@fkorotkov: if you're still collecting data about sudden test terminations, I ran into a few instances here:
https://cirrus-ci.com/task/5451118973812736
https://cirrus-ci.com/task/6014068927234048

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.End));
+
+    await tester.pumpAndSettle();
+    // The cursor has been set.
+    expect(controller.selection.baseOffset, 8);
+
+    // Go in the other direction.
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.Start));
+    // Sets the origin.
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.Update,
```
Comment (goderbauer) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart]: about these indentations: I think our style guide wants the closing ) on the same indentation as the beginning of the statement.

**Review by goderbauer — APPROVED** [review]
LGTM modulo some indentation nits.

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    editable.selection = const TextSelection(baseOffset: 29, extentOffset: 29);
+
+    final EditableTextState editableTextState = tester.firstState(find.byType(EditableText));
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.Start));
+    editableTextState.updateFloatingCursor(RawFloatingCursorPoint(state: FloatingCursorDragState.Update,
+      offset: const Offset(20, 20)));
+    await tester.pump();
+
+    expect(find.byType(EditableText), paints..rrect(
+      rrect: RRect.fromRectAndRadius(Rect.fromLTRB(464.5, 0, 467.5, 16.0), const Radius.circular(1.0)), color: const Color(0xff4285f4))
```
Comment (jslavitz) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart]: Don't worry about the indentation here as I fix it in my PR.

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+          cursorRadius: const Radius.circular(3.0),
+        ),
+      ),
+    );
+    await tester.pumpWidget(widget);
+
+    // Populate a fake clipboard.
+    const String clipboardContent = ' ';
+    SystemChannels.platform
+        .setMockMethodCallHandler((MethodCall methodCall) async {
```
Comment (jslavitz) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart]: nit: fix indentation here

**Review by jslavitz — APPROVED** [review]
LGTM

**Comment by xster** [comment]
Some wild indents snuck into the copypasta. Fixed

**Comment by goderbauer** [comment]
Looks like Cirrus is not happy with this PR.

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+    await tester.tap(find.byType(TextField));
+    await tester.pump();
+
+    final EditableTextState editableTextState = tester.firstState(find.byType(EditableText));
+    final RenderEditable renderEditable = editableTextState.renderEditable;
+
+    expect(renderEditable.cursorColor.alpha, 255);
+
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 200));
```
Comment (xster) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart]: @jslavitz I tweaked this a bit. In general, we want to be very with the number and durations of pumps to make sure the animation happens exactly the way we expect it to.

**Pull request merged**